### PR TITLE
[FMV][AArch64] Expand feature dependencies using AArch64::ExtensionSet.

### PR DESCRIPTION
**Pull request by labrinea**
Currently we maintain a hand written list of subtarget features which we are implied for a given FMV feature. It is more robust to expand such dependencies using ExtensionDependency from TargetParser, since that is generated by tablegen. For this to work each FMV feature must have a corresponding SubtargetFeature in place. FMV features which didn't satisfy this criteria have been removed from the ACLE specification (https://github.com/ARM-software/acle/pull/315). However, I deliberately marked the ArchExtKind in FMVInfo structure as std::optional in case we decide to break this rule in the future.

I have also added the missing dependencies:
 * FEAT_DPB2 -> FEAT_DPB
 * FEAT_FlagM2 -> FEAT_FlagM

### CODE REVIEW

File: llvm/include/llvm/TargetParser/AArch64TargetParser.h [hunk 1]
```diff
@@ -68,19 +68,13 @@ struct ExtensionInfo {
...
-  FMVInfo(StringRef Name, CPUFeatures Bit, StringRef Features,
+  StringRef Name;                // The target_version/target_clones spelling.
+  CPUFeatures Bit;               // Index of the bit in the FMV feature bitset.
+  std::optional<ArchExtKind> ID; // The architecture extension to enable.
+  unsigned Priority;             // FMV priority.
+  FMVInfo(StringRef Name, CPUFeatures Bit, std::optional<ArchExtKind> ID,
           unsigned Priority)
-      : Name(Name), Bit(Bit), Features(Features), Priority(Priority){};
-
-  SmallVector<StringRef, 8> getImpliedFeatures() {
```
Comment (jthackray) [inline llvm/include/llvm/TargetParser/AArch64TargetParser.h:79]: Good to see this gone, much cleaner :)

**Review by jthackray — APPROVED** [review]
Looks like a good improvement.

**Comment by labrinea** [comment]
Thanks. IMHO this counts as supporting evidence to why some FMV features are obsolete and should be unified with others or removed from the ACLE spec.

**Comment by labrinea** [comment]
I've left out the bf16 and i8mm dependencies fixup as they deserve a separate patch. I will be merging this until the end of today if no one objects.

**Review by jroelofs — APPROVED** [review]

**Pull request merged**